File: 프로그래머스/unrated/150369. 택배 배달과 수거하기/택배 배달과 수거하기.java
class Solution {
    public long solution(int cap, int n, int[] deliveries, int[] pickups) {
        long answer = 0;
        
        int deliver_n = 0;
        int pick_n = 0;
        
        for(int i=n-1; i>=0; i--){
            deliver_n -= deliveries[i];
            pick_n -= pickups[i];
            
            // 한 번 다녀왔지만 배달해야하거나 수거해야 할 물건이 남아있는 경우 또 가야하므로 while 루프 
            while(deliver_n < 0 || pick_n < 0) { 
                deliver_n += cap;
                pick_n += cap; 
                answer += (i+1)*2;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/unrated/150369. 택배 배달과 수거하기/택배 배달과 수거하기.py
def solution(cap, n, deliveries, pickups):
    answer = 0
    
    deli_n = 0
    pick_n = 0
    for i in range(n-1, -1, -1):
        deli_n -= deliveries[i]
        pick_n -= pickups[i]
        
        while deli_n < 0 or pick_n < 0:
            deli_n += cap
            pick_n += cap
            answer += (i+1)*2
    
    return answer